### PR TITLE
Customer dashboard

### PR DESCRIPTION
**Pull request by kjugi**
https://dev.snowdog.pro/issues/31973

## Add templates to modules:
+Magento_AdvancedCheckout
+Magento_Customer
+Magento_CustomerBalance
+Magento_CustomerCustomAttributes
+Magento_Downloadable
+Magento_GiftCardAccount
+Magento_GiftMessage
+Magento_GiftRegistry
+Magento_Invitation
+Magento_Review
+Magento_Reward
+Magento_Sales
+Magento_Shipping
+Magento_Vault
+Magento_Wishlist

Change formatting in files to our, force BEM methodology, create few JS script files like `theme-frontend-alpaca/Magento_Customer/web/js/countryList.babel.js ` which manipulate with country and region selects.

### CODE REVIEW

**Comment by Igloczek** [comment]
It's hard to write same think in almost every file so I'll describe common issues right there:
* missing new lines on the end of files
* all of `echo` should be replaced by `<?=`
* every `;?>` and `:?>` need to be replaced by `; ?>` and `: ?>`, b/c before `?>` we need a space
* check all `data-mage-init`, b/c many of them looks like a spaghetti, not a readable code
* do not keep opening `{`. This is how it should looks like:
   ```
   data-mage-init='{
       "something": {
           "next": true
       }
   }'
    ```
* tack down extremely long lines and try to do something with them, i.e. move values to variables first and then make something with it

File: Magento_GiftRegistry/templates/edit/registrants.phtml [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                    break;
+
+                case 'select':
+                    $options = [];
+                    if (is_array($data['options'])) {
+                        foreach ($data['options'] as $option) {
+                            $options[] = ['label' => $option['label'], 'value' => $option['code']];
+                        }
+                    }
+        ?>
```
[the remark this answers is not in the document]
Comment (kjugi) [reply Magento_GiftRegistry/templates/edit/registrants.phtml]: It was been in backend dev. Think they will make module from it but said it's ok 😕

File: Magento_GiftRegistry/templates/edit/registry.phtml [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                               class="input__field required-entry"
+                               name="<?php /* @escapeNotVerified */ echo $id; ?>"
+                               id="<?php /* @escapeNotVerified */ echo $id; ?>"
+                        />
+                    </div>
+        <?php
+                    break;
+
+                default:
+        ?>
```
[the remark this answers is not in the document]
Comment (kjugi) [reply Magento_GiftRegistry/templates/edit/registry.phtml]: It was been in backend dev. Think they will make module from it but said it's ok 😕

File: Magento_AdvancedCheckout/templates/customer/file.phtml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<div class="file-upload skus">
+    <label class="file-upload__label" for="customer_sku_csv">
+        <?= /* @escapeNotVerified */ __('File extensions allowed: .csv'); ?>
+    </label>
+
+    <div class="file-upload__handler" 
+         data-mage-init='{
+             "fieldsetControls": {}, 
+             "fileChooser": {
+                 "fileNameSelector": "[name=<?= /* @escapeNotVerified */ \Magento\AdvancedCheckout\Model\Import::FIELD_NAME_SOURCE_FILE ?>]",
```
[the remark this answers is not in the document]
Comment (iTymz) [reply Magento_AdvancedCheckout/templates/customer/file.phtml]: Yes ;) You can use PHP aliases to import those classes as the easiest way
`use Magento\AdvancedCheckout\Model\Import as ImportModel`

[the remark this answers is not in the document]
Comment (kjugi) [reply Magento_AdvancedCheckout/templates/customer/file.phtml]: thanks, done

**Pull request merged**